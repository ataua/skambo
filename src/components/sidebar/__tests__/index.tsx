import React from "react";
import renderer from "react-test-renderer";
import Sidebar from "../index";
import "jest-styled-components";

import "mutationobserver-shim";

global.MutationObserver = window.MutationObserver;

describe("Sidebar", () => {
  it("Should render sidebar", () => {
    const tree = renderer.create(<Sidebar />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
