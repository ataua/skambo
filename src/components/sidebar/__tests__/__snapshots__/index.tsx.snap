// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sidebar Should render sidebar 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  width: 300px;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  margin-top: 50px;
}

.c0 h3 {
  text-align: left;
  padding: 10px;
  width: 200px;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 200px;
  font-size: 18px;
  padding: 10px;
  border-radius: 15px;
  cursor: pointer;
}

@media (max-width:480px) {
  .c0 {
    display: none;
  }
}

<div
  className="c0"
>
  <h3>
     Categorias 
  </h3>
  <div
    className="c1 games"
  >
    <svg
      className="icon-style"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M15.4695 11.2929C15.0789 10.9024 14.4458 10.9024 14.0553 11.2929C13.6647 11.6834 13.6647 12.3166 14.0553 12.7071C14.4458 13.0976 15.0789 13.0976 15.4695 12.7071C15.86 12.3166 15.86 11.6834 15.4695 11.2929Z"
        fill="currentColor"
      />
      <path
        d="M16.1766 9.17156C16.5671 8.78103 17.2003 8.78103 17.5908 9.17156C17.9813 9.56208 17.9813 10.1952 17.5908 10.5858C17.2003 10.9763 16.5671 10.9763 16.1766 10.5858C15.7861 10.1952 15.7861 9.56208 16.1766 9.17156Z"
        fill="currentColor"
      />
      <path
        d="M19.7121 11.2929C19.3216 10.9024 18.6885 10.9024 18.2979 11.2929C17.9074 11.6834 17.9074 12.3166 18.2979 12.7071C18.6885 13.0976 19.3216 13.0976 19.7121 12.7071C20.1027 12.3166 20.1027 11.6834 19.7121 11.2929Z"
        fill="currentColor"
      />
      <path
        d="M16.1766 13.4142C16.5671 13.0237 17.2003 13.0237 17.5908 13.4142C17.9813 13.8048 17.9813 14.4379 17.5908 14.8284C17.2003 15.219 16.5671 15.219 16.1766 14.8284C15.7861 14.4379 15.7861 13.8048 16.1766 13.4142Z"
        fill="currentColor"
      />
      <path
        d="M6 13H4V11H6V9H8V11H10V13H8V15H6V13Z"
        fill="currentColor"
      />
      <path
        clipRule="evenodd"
        d="M7 5C3.13401 5 0 8.13401 0 12C0 15.866 3.13401 19 7 19H17C20.866 19 24 15.866 24 12C24 8.13401 20.866 5 17 5H7ZM17 7H7C4.23858 7 2 9.23858 2 12C2 14.7614 4.23858 17 7 17H17C19.7614 17 22 14.7614 22 12C22 9.23858 19.7614 7 17 7Z"
        fill="currentColor"
        fillRule="evenodd"
      />
    </svg>
     
    Jogos
  </div>
  <div
    className="c1 books"
  >
    <svg
      className="icon-style"
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M4 19.5A2.5 2.5 0 0 1 6.5 17H20"
      />
      <path
        d="M6.5 2H20v20H6.5A2.5 2.5 0 0 1 4 19.5v-15A2.5 2.5 0 0 1 6.5 2z"
      />
    </svg>
     
    Livros
  </div>
  <div
    className="c1 clothes"
  >
    <svg
      className="icon-style"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 3a3 3 0 0 0 6 0h6a1 1 0 0 1 1 1v7a1 1 0 0 1-1 1h-2.001L19 20a1 1 0 0 1-1 1H6a1 1 0 0 1-1-1l-.001-8.001L3 12a1 1 0 0 1-1-1V4a1 1 0 0 1 1-1h6zm11 1.999h-3.417l-.017.041a5.002 5.002 0 0 1-4.35 2.955L12 8a5.001 5.001 0 0 1-4.566-2.96L7.416 5H4v5l2.999-.001V19H17l-.001-9L20 9.999v-5z"
          fillRule="nonzero"
        />
      </g>
    </svg>
     
    Vestu√°rios
  </div>
  <div
    className="c1 bike"
  >
    <svg
      className="icon-style"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M316.695 89.836c-7.041 5.311-30.985 16.92-61.937 18.803l1.094 17.965c18.246-1.11 33.904-4.842 46.37-9.4l11.755 28.087c-5.284 27.528-17.35 66.21-33.762 97.492-8.143 11.525-28.555 46.051-40.477 54.319 10.815.008 19.976 5.61 25.61 13.087a38.883 38.883 0 0 1 2.359 3.536l78.668-91.022 7.658 18.3c-30.986 15.903-52.25 48.19-52.25 85.325 0 52.824 43.024 95.836 95.858 95.836 52.833 0 95.859-43.012 95.859-95.836 0-52.823-43.026-95.836-95.86-95.836a95.52 95.52 0 0 0-26.884 3.846l-38.057-90.934 11.045-1.045c1.75-3.507 5.703-11.74-3.119-18.713l-15.885.74-6.088-14.544c3.502-1.924 6.495-3.837 8.881-5.637zm-153.617 66.543c-7.176-.031-13.677.695-18.119 2.69-6.603 2.964-4.67 21.155 5.441 27.327 7.192.44 13.733-.15 21.026-1.314l4.678 11.367H60.072v17.998H73.31l4.16 14.557C42.879 243.488 18.5 277.686 18.5 317.447c0 52.824 43.026 95.838 95.86 95.838 35.996 0 67.433-19.97 83.818-49.402l-19.278-2.797c-13.976 20.659-37.63 34.2-64.54 34.2-43.11 0-77.862-34.745-77.862-77.839 0-31.71 18.83-58.88 45.947-71.031l15.215 53.24c2.9-1.577 6.224-2.554 9.791-2.554h8.197l-16.03-56.09a78.697 78.697 0 0 1 14.741-1.403c13.051 0 25.325 3.202 36.112 8.838l-31.692 48.655h21.479l25.094-38.522c11.54 9.993 20.096 23.327 24.185 38.522h18.494c-4.652-21.406-16.477-40.176-32.83-53.643l12.408-19.049 29.916 72.692h19.809l-47.83-115.383c5.566-1.076 11.77-2.239 18.976-3.399 15.623-2.514 10.974-10.972-.351-15.304-3.976-1.521-26.712-6.557-45.05-6.637zm162.772 17.28l12.763 30.5-45.564 52.718c1.05-1.89 2.088-3.798 3.103-5.732 12.935-24.655 23.022-52.67 29.698-77.487zM92.025 214.446h80.592l-12.32 18.916c-13.65-7.487-29.308-11.752-45.938-11.752-6.751 0-13.34.706-19.703 2.041zm305.616 34.043c43.109 0 77.86 34.744 77.86 77.838.001 43.094-34.751 77.838-77.86 77.838-43.11 0-77.86-34.744-77.86-77.838 0-29.828 16.656-55.646 41.201-68.72l28.358 67.755 16.603-6.949-28.195-67.367a78.382 78.382 0 0 1 19.893-2.557zM107.45 315.1c-.958 0-1.593.253-2.71 1.636-1.118 1.384-2.159 3.885-2.38 6.399-.22 2.514.327 4.86 1.303 6.386.972 1.52 2.157 2.632 5.404 3.245L231.273 350.5l.14.025c10.011 1.771 15.404-.189 18.696-2.888 3.293-2.7 5.112-6.94 5.287-12.06.176-5.121-1.548-10.74-4.423-14.557-2.876-3.817-6.29-5.92-11.258-5.92z"
      />
    </svg>
     
    Bicicletas
  </div>
  <div
    className="c1 sports"
  >
    <svg
      className="icon-style"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M19.071,4.929c-1.895-1.895-4.405-2.938-7.07-2.938c-2.666,0-5.177,1.044-7.072,2.938c-3.899,3.898-3.899,10.243,0,14.142 c1.895,1.895,4.406,2.938,7.073,2.938c2.666,0,5.176-1.043,7.07-2.937C22.971,15.174,22.971,8.829,19.071,4.929z M12.181,4h-0.359 c0.061-0.001,0.119-0.009,0.18-0.009S12.12,3.999,12.181,4z M18.243,17H16l-1.258,2.516c-0.868,0.318-1.791,0.493-2.741,0.493 c-0.952,0-1.876-0.175-2.746-0.494L8,17.01H5.765c-0.846-1.052-1.387-2.269-1.623-3.532L6,11L4.784,8.567 c0.381-0.804,0.895-1.56,1.559-2.224c0.919-0.92,2.021-1.584,3.22-1.969L12,6l2.438-1.625c1.198,0.385,2.301,1.049,3.22,1.968 c0.663,0.663,1.177,1.418,1.558,2.221L18,11l1.858,2.478C19.623,14.737,19.085,15.95,18.243,17z"
      />
      <path
        d="M8.5 11L10 15 14 15 15.5 11 12 8.5z"
      />
    </svg>
     
    Esportes
  </div>
  <div
    className="c1 instruments"
  >
    <svg
      className="icon-style"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M502.63 39L473 9.37a32 32 0 0 0-45.26 0L381.46 55.7a35.14 35.14 0 0 0-8.53 13.79L360.77 106l-76.26 76.26c-12.16-8.76-25.5-15.74-40.1-19.14-33.45-7.78-67-.88-89.88 22a82.45 82.45 0 0 0-20.24 33.47c-6 18.56-23.21 32.69-42.15 34.46-23.7 2.27-45.73 11.45-62.61 28.44C-16.11 327-7.9 409 47.58 464.45S185 528 230.56 482.52c17-16.88 26.16-38.9 28.45-62.71 1.76-18.85 15.89-36.13 34.43-42.14a82.6 82.6 0 0 0 33.48-20.25c22.87-22.88 29.74-56.36 22-89.75-3.39-14.64-10.37-28-19.16-40.2L406 151.23l36.48-12.16a35.14 35.14 0 0 0 13.79-8.53l46.33-46.32a32 32 0 0 0 .03-45.22zM208 352a48 48 0 1 1 48-48 48 48 0 0 1-48 48z"
      />
    </svg>
     
    Instrumentos Musicais
  </div>
  <div
    className="c1 furniture"
  >
    <svg
      className="icon-style"
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 512 512"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M112 73c-13.75 0-24.214 4.87-33.047 13.271-8.832 8.402-15.755 20.6-20.414 34.575-5.887 17.661-7.953 38.064-6.887 56.111a57.246 57.246 0 0 1 2.461-.187c8.38-.405 16.317 1.094 23.494 4.234 14.355 6.28 25.436 18.27 34.057 32.28 15.104 24.543 23.187 55.746 24.96 80.585 114.097-14.223 124.655-14.223 238.753 0 1.772-24.84 9.855-56.042 24.959-80.586 8.62-14.009 19.702-25.999 34.057-32.28 7.875-3.444 16.67-4.898 25.955-4.046 1.066-18.047-1-38.45-6.887-56.111-4.659-13.976-11.582-26.173-20.414-34.575C424.214 77.87 413.75 73 400 73zM56.969 194.502c-.686-.008-1.38.006-2.082.04-5.621.284-11.8 1.978-18.862 5.509-23.413 11.707-26.349 65.687 14.82 79.41l4.505 1.502L86.79 391h338.42l31.44-110.037 4.504-1.502c41.17-13.723 38.234-67.703 14.82-79.41-14.121-7.061-24.722-6.775-34.367-2.555-9.645 4.22-18.564 13.23-25.943 25.22C400.906 246.7 393 282 393 304v10.195l-10.117-1.265c-128-16-125.766-16-253.766 0L119 314.195V304c0-22-7.906-57.3-22.664-81.283-7.38-11.991-16.298-21.001-25.943-25.22-4.22-1.847-8.624-2.94-13.424-2.995zM83.727 409l-23.272 23.271c11.994 3.239 33.75 3.763 42.545-1.666 3.272-4.695 6.903-14.536 10.438-21.605zm314.836 0c1.494 6.017 6.952 19.12 10.437 21.605 13.2 5.463 29.904 5.774 42.545 1.666L428.273 409z"
      />
    </svg>
     
    M√≥veis
  </div>
  <div
    className="c1"
  >
    <h5>
       Ver todas as categorias 
    </h5>
  </div>
</div>
`;
